### DISCUSSION TITLE
[Projects Beta] Add iteration filtering options

### DISCUSSION THREAD

**Discussion by joeppeeters** [community/community#10862] [Projects and Issues] [answered]
We have an Issue Table grouped by iteration with a look-a-head for upcoming work. However, passed iterations are listed there as well which are not relevant. Currently the only way to filter them out is using a manual exclusion (`-`).

It would be great to have more iteration filtering like its possible with `@current`.

Suggestions:
 - `@planned` / `@upcoming` to only show all future iterations
 - `@previous` to show the most recent iteration
 - `@next` to show the next iteration
 - `@completed` for finished iterations (to be used in an exclusion: `-iteration: @completed`)

**Comment by miedzikd** [comment]
This functionality is already there for `iteration:@previous` `iteration:@next`... For `@planned` you can use `-iteration:"@previous"` - not previous iteration for

**Reply by joeppeeters** [reply]
I didn't know about those. Thanks!

However, `@previous` and `@next` seem to refer to a single iteration, while I'm looking for a way to filter on all upcoming or completed iterations. Hence the `@completed`/`@planned` proposals

**Reply by miedzikd** [reply]
I like these ideas... 
For all upcoming iterations you can use the negation of previous `-iteration:"@previous"`. This will show you current and next iterations...
but `@completed` - this one is hard to get with current options

**Reply by chrisglein** [reply]
Note that if you do filter out old iterations, the items are hidden from the board view but not the actual column for that iteration (it shows up but is empty). Is that a bug?

**Reply by rpsu** [reply]
I'd also like (optionally) to hide empty kanban columns such as completed iterations, instead of display them without items.

Ideally this would be configrable per view so that sprint columns (todo, in progress, done) would still be visible but sprints (sprint 1, sprint 2 etc.) kanban view could be configured to hide the sprints w/o items.

**Reply by nestor-sk** [reply]
`-iteration:"@previous"` only hides ONE previous iteration, the last one.

**Comment by nestor-sk** [comment]
We need the `iteration:"@completed"` filter only. This will allow filtering for or out the completed ones. If you want upcoming, we can do `-iteration:"@completed",@"current"`

It is quite annoying to see that the concepts are there in the settings but not in the filters
<img width="321" alt="Screenshot 2022-05-31 at 17 53 30" src="https://user-images.githubusercontent.com/95345919/171216917-e4cd6d68-b418-4986-8742-5922ca75e1ec.png">

**Reply by tayloramurphy** [reply]
Big +1 on this. Having to delete iterations to get them to drop from the board view is disappointing as you lose all of that history.

**Comment by mavam** [comment, marked as answer]
This worked for me in the table view: `iteration:>=@current`. The key is `>=` to select a range of issues.

There's also now arithmetic available, e.g., `iteration:<=@current+3` for everything up to the next 3 iterations only.

NB: unfortunately this doesn't hide done/empty iterations in the Board View. This seems to be [a general issue](https://github.com/community/community/discussions/6906), though.

**Reply by joeppeeters** [reply]
Perfect answer, thanks. 

The docs (now?) also seem to include it: https://docs.github.com/en/issues/planning-and-tracking-with-projects/customizing-views-in-your-project/filtering-projects#filtering-items

**Comment by mlangenberg** [comment]
Thanks! Using `iteration:>=@previous-1` as a filter makes it easy to reference issues from the two previous iterations and the active one. Still I wished that completed iterations would be collapsed by default when used as a group.
